builder/replica_pipeline.cc

```diff
@@ -146,8 +147,25 @@ class pipeline_replicator : public expr_visitor {
     buffers_emitted_.insert(bep->sym());
 
     std::string elem_size = print_expr_inlined(bep->elem_size());
-    (void)print_assignment_explicit(
-        name, "buffer_expr::make(ctx, \"", name, "\", /*rank=*/", bep->rank(), ", /*elem_size=*/", elem_size, ")");
+    if (bep->constant()) {
+      std::string const_name = name + "_const";
+      (void)print_assignment_explicit(
+          const_name, "std::make_shared<buffer<void, ", bep->rank(), ">>(/*rank=*/", bep->rank(), ", /*elem_size=*/", elem_size, ")");
+      for (std::size_t d = 0; d < bep->rank(); d++) {
+          std::string m = print_expr_inlined(bep->dim(d).bounds.min);
+          std::string e = print_expr_inlined(bep->dim(d).bounds.extent());
+          os_ << "  " << const_name << "->dim(" << d << ").set_min_extent(" << m << ", " << e << ");\n";
```

Comment:
It would be better to use `set_bounds(min, max)`, which will always work (vs. min-extent which can overflow).